File: lab_1/Matrix.cs
using System;

namespace Mat
{
    public class Matrix<T>
    {
        T[, ] matrix;
        int column;
        int row;

        Matrix(int row, int column)
        {
            this.row = row;
            this.column = column;
            T[, ] matrix = new T[row, column];
        }

        Matrix(T[, ] matrix, int row, int column)
        {
            this.matrix = matrix;
            this.row = row;
            this.column = column;
        }

        public int Row 
        {
            get
            {
                return row;
            }
        }

        public int Column
        {
            get
            {
                return column;
            }
        }
    }
}

File: IvanLyuhtikov/Task1/matrix/src/main/java/entity/Matrix.java
package entity;

/**
 * Simple representation of 2d matrix, stores a state only!
 * @param <T> - generic type
 */
public class Matrix<T> {
    private T[][] array2d;
    private int col;
    private int row;

    public Matrix(T[][] array2d, int col, int row) {
        this.array2d = array2d;
        this.col = col;
        this.row = row;
    }

    public T[][] getMatrix() {
        return this.array2d;
    }

    public int getCol() {
        return this.col;
    }

    public int getRow() {
        return this.row;
    }

    public T get(int i, int j) {
        return this.array2d[i][j];
    }

    public void setSimpleValue(T val, int i, int j) {
        this.array2d[i][j] = val;
    }
}
